File: docs/guides/php/src/searchWithLogicalOr.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\OptionalWords;
use Algolia\AlgoliaSearch\Model\Search\SearchParamsObject;

$client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

$query = 'the query';

$optionalWords = new OptionalWords();
$optionalWords[] = $query;

$searchParams = (new SearchParamsObject())
    ->setQuery($query)
    ->setOptionalWords($optionalWords)
;

$client->searchSingleIndex(
    '<YOUR_INDEX_NAME>',
    $searchParams,
);

File: docs/guides/ruby/searchWithLogicalOr.rb
require "algolia"

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

query = "the query"
optional_words = %w[the query]
search_params = Algolia::Search::SearchParamsObject.new(
  query: query,
  optional_words: optional_words
)
client.search_single_index("<YOUR_INDEX_NAME>", search_params)
